db/migrate/20210119082215_add_nickname_to_users.rb

```diff
@@ -0,0 +1,6 @@
+class AddNicknameToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :nickname, :string, null: true
+    add_index :users, :nickname, unique: true, where: 'nickname IS NOT NULL'
```

Comment:
please add tests for this case

1) try to insert 2 user with same nickname
2) try to insert 2 user without nickname (check if there is no validation in uniq for such case)